modules/hitl2/src/backend/repository.ts

```diff
@@ -0,0 +1,367 @@
+        `comments.createdAt as ${this.commentPrefix}:createdAt`
+      )
+      .leftJoin('comments', 'escalations.id', 'comments.escalationId')
+      .andWhere('escalations.botId', botId)
+      .distinct()
+      .modify(this.applyLimit, limit)
+      .modify(this.applyOrderBy, orderByColumn, orderByDirection)
+      .orderBy([{ column: 'comments.createdAt', order: 'asc' }])
+  }
+
+  private applyQuery = (query?: Knex.QueryCallback) => {
+    return (builder: Knex.QueryBuilder) => {
+      if (query) {
+        return builder.modify(query)
+      } else {
+        return builder
+      }
+    }
+  }
+
+  escalationsQuery = async (query?: Knex.QueryCallback): Promise<EscalationType[]> => {
+    return await this.bp.database<EscalationType>('escalations').modify(this.applyQuery(query))
+  }
+
+  getAgentOnline = async (botId: string, agentId: string): Promise<boolean> => {
+    const value = await this.bp.kvs.forBot(botId).get(`hitl2:online:${agentId}`)
+    return !!value
+  }
+
+  setAgentOnline = async (botId: string, agentId: string, value: boolean): Promise<boolean> => {
+    const config = await this.bp.config.getModuleConfigForBot('hitl2', botId)
+    await this.bp.kvs.forBot(botId).set(`hitl2:online:${agentId}`, value, null, config.agentSessionTimeout)
+    return value
+  }
+
+  getCurrentAgent = async (req, botId: string, agentId: string): Promise<AgentType> => {
+    const online = await this.getAgentOnline(botId, agentId)
+
+    return axios
+      .get('/auth/me/profile', await this.axiosConfig(req, botId))
+      .then(response => response.data.payload)
+      .then(data => {
+        return {
+          ...data,
+          id: agentId,
+          online: online
+        } as AgentType
+      })
+  }
+
+  getAgents = async (botId: string, conditions: AgentCollectionConditions = {}): Promise<AgentType[]> => {
+    let { online } = conditions
+
+    const applyConditions = (records: []) => {
+      if ('online' in conditions) {
+        return _.filter<AgentType>(records, ['online', online])
+      } else {
+        return records
+      }
+    }
+
+    return await this.bp
+      .database('workspace_users')
+      .then(data => {
+        return data.map(async row => {
+          return {
+            ...row,
+            id: makeAgentId(row.strategy, row.email),
+            online: await this.getAgentOnline(botId, makeAgentId(row.strategy, row.email))
+          }
+        })
+      })
+      .then(async agents => {
+        const superAdmins = (await this.bp.config.getBotpressConfig()).superAdmins
+        const admins = superAdmins.map(async admin => {
+          return {
+            ...admin,
+            id: makeAgentId(admin.strategy, admin.email),
+            online: await this.getAgentOnline(botId, makeAgentId(admin.strategy, admin.email)),
+            workspace: 'default'
+          }
+        })
+
+        return Promise.all([...agents, ...admins])
+      })
+      .then(applyConditions)
```

Comment:
Same here, please use await instead. Also, this seems to do a lot of things, which is not very clear. 
What do you think of this? 
```js
const onlineUsers = await .....
const onlineSuperAdmins = await ...

return [...onlineUsers, onlineSuperAdmins].filter()